src/app/community/_components/PostCardDetailModal.tsx

```diff
@@ -1,111 +1,259 @@
     }
+    queryClient.invalidateQueries({ queryKey: ['postCardsList'] });
+    await refetch();
   };
 
+  const { mutate: postCommentMutation } = useMutation({
+    mutationFn: postComment,
+    onSuccess: () => {
+      handleSuccessSubmitComment();
+    },
+    onError: () => {
+      toast.error('댓글 등록 중 오류가 발생하였습니다.');
+    },
+  });
+
+  const { mutate: deletePostMutation } = useMutation({
+    mutationFn: deletePost,
+    onSuccess: () => {
+      toast.success('리뷰를 삭제되었습니다.');
+      queryClient.invalidateQueries({
+        queryKey: ['MyCustomReview'],
+      });
+    },
+    onError: () => {
+      toast.error('리뷰 삭제 중 오류가 발생하였습니다.');
+    },
+  });
+
   useEffect(() => {
-    const removeEvent = addEnterKeyEvent({ element: commentRef, callback: handleSubmitComment });
+    const handleSubmitComment = () => {
+      if (commentRef) {
+        const commentContent = commentRef.value;
+        postCommentMutation({ id: cardId, content: commentContent });
+      }
+    };
+    const removeEvent = addEnterKeyEvent({ element: { current: commentRef }, callback: handleSubmitComment });
+
     return () => {
       removeEvent();
     };
-  }, []);
+  }, [cardId, postCommentMutation, commentRef]);
+
+  if (isPending) {
+    return <span>Loading...</span>;
+  }
+
+  if (isError) {
+    toast.error('데이터를 불러오는데 실패하였습니다.');
+    return <span>Error: {error.message}</span>;
+  }
+
+  const { data: postData, status, message } = data;
+
+  if (status === 'FAIL') {
+    toast.error(message);
+    onClose();
+    return null;
+  }
+
+  const { commentCount, comments, content, likeCount, nickName, reviewImages, title, updatedAt, userImage, custom } =
+    postData as CommunityPostCardDetailDataType;
 
-  const handleThumbnailClick = (i: number) => {
-    setClickedImage(keyboardImages[i].src);
+  // const { type, texture, boardColor, switchType, baseKeyColor, hasPointKeyCap, pointKeyType, individualColor } = custom;
+
+  const createdDateString = formatDateToString(new Date(updatedAt));
+
+  // const options = {
+  //   type,
+  //   texture,
+  //   boardColor,
+  //   switchType,
+  //   baseKeyColor,
+  //   hasPointKeyCap,
+  //   pointKeyType,
+  //   individualColor,
+  // };
+
+  const handleClickThumbnail = (i: number) => {
+    setClickedImage(reviewImages[i].imgUrl);
+  };
+
+  const handleClickDeleteAlertButon = () => {
+    deletePostMutation(cardId);
+    setIsCheckDeleteAlertOpen(false);
+  };
+
+  const handleCloseDeleteAlert = () => {
+    setIsCheckDeleteAlertOpen(false);
+  };
+
+  const handleClickEditAlertButton = () => {
+    setIsEditModalOpen(true);
+    setIsCheckEditAlertOpen(false);
+  };
+
+  const handleCloseEditAlert = () => {
+    setIsCheckEditAlertOpen(false);
+  };
+
+  const handleClickEditModalButton = () => {
+    setIsEditModalOpen(false);
+    queryClient.invalidateQueries({
+      queryKey: ['MyCustomReview'],
+    });
+    queryClient.invalidateQueries({
+      queryKey: ['postData', cardId],
+    });
```

Comment:
되는지는 모르겠지만, 배열로 묶어서 수행할 수 있지 않을까요?
```suggestion
    queryClient.invalidateQueries({
      queryKey: [['MyCustomReview'], ['postData', cardId]],
    })
```

Reply:
이것은 추후에 함 확인해보겠슴다,, 바꿨다가 안될 수도 잇을 것 같아서욤,,